packages/world/src/systems/NatureSystem.sol

```diff
@@ -77,128 +74,10 @@ contract NatureSystem is System {
-
-      uint128 growableEnergy = objectType.getGrowableEnergy();
-      uint128 trunkEnergy = trunkHeight * ObjectPhysics._getEnergy(treeData.logType);
-      uint128 leafEnergy = leaves * ObjectPhysics._getEnergy(treeData.leafType);
-
-      uint128 energyToReturn = growableEnergy - trunkEnergy - leafEnergy;
-
-      if (energyToReturn > 0) {
-        addEnergyToLocalPool(coord, energyToReturn);
-      }
-    }
-  }
-
-  function _growTree(EntityId seed, Vec3 baseCoord, TreeData memory treeData, ObjectType saplingType)
-    private
-    returns (uint32, uint32)
-  {
-    uint32 trunkHeight = _growTreeTrunk(seed, baseCoord, treeData);
-
-    if (trunkHeight <= 2) {
-      // Very small tree, no leaves
-      return (trunkHeight, 0);
-    }
-
-    // Adjust if the tree is blocked
-    bool obstructed = trunkHeight < treeData.trunkHeight;
-    if (obstructed) {
-      trunkHeight = trunkHeight + 1; // Still allow one layer above the trunk
-    }
-
-    (Vec3[] memory fixedLeaves, Vec3[] memory randomLeaves) = TreeLib.getLeafCoords(saplingType);
-
-    // Initial seed for randomness
-    uint256 rand = uint256(keccak256(abi.encodePacked(block.timestamp, baseCoord)));
-
-    uint32 leafCount;
-
-    for (uint256 i = 0; i < fixedLeaves.length; ++i) {
-      Vec3 rel = fixedLeaves[i];
-      if (obstructed && rel.y() > int32(trunkHeight)) {
-        break;
-      }
-
-      if (_tryCreateLeaf(treeData.leafType, baseCoord + rel)) {
-        ++leafCount;
-      }
-    }
-
-    for (uint256 j = 0; j < randomLeaves.length; ++j) {
-      Vec3 rel = randomLeaves[j];
-      if (obstructed && rel.y() > int32(trunkHeight)) {
-        break;
-      }
-
-      rand = uint256(keccak256(abi.encodePacked(rand, j))); // evolve RNG
-
-      if (rand % 100 < 40) continue; // 40 % trimmed
-
-      if (_tryCreateLeaf(treeData.leafType, baseCoord + rel)) {
-        ++leafCount;
-      }
-    }
-
-    return (trunkHeight, leafCount);
-  }
-
-  function _tryCreateLeaf(ObjectType leafType, Vec3 coord) private returns (bool) {
-    (EntityId leaf, ObjectType existing) = EntityUtils.getOrCreateBlockAt(coord);
-    if (existing != ObjectTypes.Air) {
-      return false;
-    }
-
-    EntityObjectType._set(leaf, leafType);
-    return true;
-  }
-
-  function _growTreeTrunk(EntityId seed, Vec3 baseCoord, TreeData memory treeData) private returns (uint32) {
-    // Replace the seed with the trunk
-    EntityObjectType._set(seed, treeData.logType);
-
-    // Create the trunk up to available space
-    for (uint32 i = 1; i < treeData.trunkHeight; i++) {
-      Vec3 trunkCoord = baseCoord + vec3(0, int32(i), 0);
-      (EntityId trunk, ObjectType objectType) = EntityUtils.getOrCreateBlockAt(trunkCoord);
-      if (objectType != ObjectTypes.Air) {
-        return i;
-      }
-
-      EntityObjectType._set(trunk, treeData.logType);
-    }
-
-    return treeData.trunkHeight;
+    // TODO: should we do proximity checks?
```

Comment:
I dont think proximity checks are needed cuz this is just updating the onchain state to reflect what the optimistic value is, cuz theres no tick onchain